lib/Raven/ClientBuilder.php

```diff
@@ -104,12 +128,40 @@ public static function create(array $options = [])
         return new static($options);
     }
 
+    public function setMessageFactory(MessageFactory $messageFactory)
+    {
+        $this->messageFactory = $messageFactory;
+    }
+
+    public function setStreamFactory(StreamFactory $streamFactory)
+    {
+        $this->streamFactory = $streamFactory;
+    }
+
+    /**
+     * Sets the factory to use to create the transport.
+     *
+     * @param TransportFactoryInterface $transportFactory The transport factory
+     */
+    public function setTransportFactory(TransportFactoryInterface $transportFactory)
+    {
+        $this->transportFactory = $transportFactory;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function getClient()
     {
-        return new Client($this->configuration);
+        $messageFactory = $this->messageFactory ?: MessageFactoryDiscovery::find();
+        $transportFactory = $this->transportFactory ?: new CurlTransportFactory(
+            $messageFactory,
+            $this->streamFactory ?: StreamFactoryDiscovery::find()
+        );
+
+        $transportFactory = new PluginClientFactory($this->configuration, $transportFactory, UriFactoryDiscovery::find());
+
+        return new Client($this->configuration, $transportFactory->getInstance(), $messageFactory);
```

Comment:
How do I instance some other class which extends \\Raven\\Client? Will I have to extend this class too?